frontend/src/pages/History.jsx

```diff
@@ -0,0 +1,372 @@
+import axios from 'axios';
+import { Link, useParams, useLocation } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+import "../css/History_style.css";
+
+export const History= () => {
+
+  const { type, name } = useParams();
+  const [ lpvsHistories, setlpvsHistories ] = useState();
+  const navigate = useNavigate();
+
+  const location = useLocation();
+  const queryParams = new URLSearchParams(location.search);
+  const page = queryParams.get('page') || 0;
+
+
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [username, setUsername] = useState("");
+
+  useEffect(() => {
+    axios.get("/login/check")
+      .then((loginresponse) => {
+        if (loginresponse.data.isLoggedIn) {
+          setIsLoggedIn(loginresponse.data.isLoggedIn);
+          axios.get("/user/info")
+            .then((userInfoResponse) => {
+              setUsername(userInfoResponse.data);
+            })
+            .catch(function(error) {
+              console.log(error.toJSON());
+              navigate("/login");
+            });
+        } else {
+          navigate("/login");
+        }
+      })
+      .catch(function(error) {
+        console.log(error.toJSON());
+        navigate("/login");
+      });
+  }, []);
+
+  useEffect(() => {
+    axios
+      .get(`/history/${type}/${name}?page=${page}`)
+      .then((response) => {
+        console.log(response.data)
+          setlpvsHistories(response.data);
+      })
+      .catch(function(error) {
+        console.log(error.toJSON());
+        const userChoice = window.confirm("Please enter the GitHub ID on the User page.");
+        if (userChoice) {
+          navigate('/user/setting');
+        } else {
+          navigate(`/history/send/${username?.nickname}?page=0`);
+        }
+      });
+  }, [type, name, page]);
+
+  const handlePageChange = (pageNumber) => {
+    navigate(`/history/${type}/${name}?page=${pageNumber}`);
+  };
+
+  const [isHistoriesEmpty, setIsHistoriesEmpty] = useState(false);
+  console.log(lpvsHistories?.lpvsHistories?.length)
+  useEffect(() => {
+    if (lpvsHistories?.lpvsHistories?.length === 0) {
+      setIsHistoriesEmpty(true);
+    } else {
+      setIsHistoriesEmpty(false);
+    }
+  }, [lpvsHistories, page]);
+  
+  console.log(lpvsHistories)
+
+  const navigateToOrg = () => {
+    if (!username?.organization || username?.organization.trim() === "") {
+      window.alert("Please enter the Organization information on the User page.");
+    } else {
+      navigate(`/history/org/${username?.organization}?page=0`);
+    }
+  };
+  const navigateToOwn = () => {
+      navigate(`/history/own/${username?.nickname}?page=0`);
+  };
+  const navigateToSend = () => {
+      navigate(`/history/send/${username?.nickname}?page=0`);
+  };
+
+   const [pageCount,setPageCount] = useState(1);
+   const [currentPage, setCurrentPage] = useState(1);
+
+   useEffect(()=> {
+    setPageCount(Math.ceil(lpvsHistories?.count/5))
```

Comment:
How about to use a flexible amount of scans that can be displayed on the page?

For example:
![image](https://github.com/Samsung/LPVS/assets/45031201/0a4fcaed-b5fa-4ed2-8d9e-5412c82c2ac3)

Reply:
@kyudori What do you think? If you need extra time for this, we can make this as an issue and implement it later.

Reply:
@o-kopysov It's possible to add a function as you reviewed it.

By the way, as @tiokim said, it will take some time will take. 

There are still two pages to upload, so it would be good to take it out as an issue and implement it later.